### ISSUE TITLE
Build errors on Windows 10/AMD64/MSYS2

### ISSUE THREAD

**Issue by kbolino** [veandco/go-sdl2#502]
# Problem

Can't build on Windows 10, AMD64/x86_64, with MSYS2 and latest versions of SDL2/go-sdl2:

```
> go build .
# github.com/veandco/go-sdl2/sdl
..\..\..\go\pkg\mod\github.com\veandco\go-sdl2@v0.4.11\sdl\render.go:1191:97: cannot use _color (type *_Ctype_int) as type *_Ctype_struct_SDL_Color in assignment
..\..\..\go\pkg\mod\github.com\veandco\go-sdl2@v0.4.11\sdl\video.go:1084:60: cannot use _size (type *_Ctype_ulong) as type *_Ctype_ulonglong in assignment
```

# Possible solution

The latter problem seems to stem from a difference between this platform and others, but can likely be resolved for all platforms by simply using the correct target type in the cast (based upon the signature of `SDL_GetWindowICCProfile`):

```c
/* replace this */
	_size := (*C.ulong)(unsafe.Pointer(&size))
/* with this */
	_size := (*C.size_t)(unsafe.Pointer(&size))
```

I can suggest a fix for the first problem but I don't understand the assumptions behind the Go signature vs. the C signature of `SDL_RenderGeometryRaw`. The C signature takes a pointer to `SDL_Color` struct while the Go signature takes a slice of `int32`. The end result is size-compatible (4 bytes either way) but why use `int32` when the library has a `Color` struct which is size-compatible also?

Anyway, the suggested fix without changing function signature would be:

```c
/* replace this */
	_color := (*C.int)(&color[0])
/* with this */
	_color := (*C.SDL_Color)(unsafe.Pointer(&color[0]))
```

This also helps with platform independence since there was another assumption here that C `int` is always 32 bits (true on most platforms nowadays, but not guaranteed).

I can submit a PR with these fixes if desired.

# Versions

```console
> ver
Microsoft Windows [Version 10.0.19043.1466]

> go list -m github.com/veandco/go-sdl2
github.com/veandco/go-sdl2 v0.4.11

> go version
go version go1.17.6 windows/amd64

$ pacman -Q | grep SDL2
mingw-w64-x86_64-SDL2 2.0.20-1
mingw-w64-x86_64-SDL2_gfx 1.0.4-1
mingw-w64-x86_64-SDL2_image 2.0.5-2
mingw-w64-x86_64-SDL2_mixer 2.0.4-5
mingw-w64-x86_64-SDL2_ttf 2.0.18-1

$ pacman -Q | grep gcc
gcc-libs 11.2.0-3
mingw-w64-x86_64-gcc 11.2.0-6
mingw-w64-x86_64-gcc-libs 11.2.0-6
```

**Comment by pendar747** [comment]
I'm seeing the same problem when trying to install on macOS big sur (v11.6.2):
```
../../go/pkg/mod/github.com/veandco/go-sdl2@v0.4.11/sdl/render.go:1191:97: cannot use _color (type *_Ctype_int) as type *_Ctype_struct_SDL_Color in assignment
```
and also
```
../../go/pkg/mod/github.com/veandco/go-sdl2@v0.4.10/ttf/sdl_ttf.go:85:32: cannot use _Ctype_int(val) (type _Ctype_int) as type _Ctype_SDL_bool in argument to _Cfunc_TTF_ByteSwappedUNICODE
```

**Comment by jncornett** [comment]
Same, I am able to reproduce on macOS.
FWIW a workaround that worked for me was to downgrade to `v0.4.9`.

Reproduction details below:

### My Environment

OS: macOS Big Sur 11.2.3
Device: MacBook Pro (M1)

```
$ go version
go version go1.17.6 darwin/arm64
```

```
$ brew ls sdl2
/opt/homebrew/Cellar/sdl2/2.0.20/bin/sdl2-config
/opt/homebrew/Cellar/sdl2/2.0.20/include/SDL2/ (77 files)
/opt/homebrew/Cellar/sdl2/2.0.20/lib/libSDL2-2.0.0.dylib
/opt/homebrew/Cellar/sdl2/2.0.20/lib/cmake/ (2 files)
/opt/homebrew/Cellar/sdl2/2.0.20/lib/pkgconfig/sdl2.pc
/opt/homebrew/Cellar/sdl2/2.0.20/lib/ (4 other files)
/opt/homebrew/Cellar/sdl2/2.0.20/share/aclocal/sdl2.m4
```

### What am I running?

In a go module:

```
$ go get github.com/veandco/go-sdl2/sdl@v0.5.0-alpha.2
# Also ran:
$ go get github.com/veandco/go-sdl2/sdl@v0.4.10
```

### What do I expect?

`go get` succeeds

### What do I see instead?

```
# github.com/veandco/go-sdl2/sdl
../../go/pkg/mod/github.com/veandco/go-sdl2@v0.5.0-alpha.2/sdl/render.go:1223:97: cannot use _color (type *_Ctype_int) as type *_Ctype_struct_SDL_Color in assignment
exit 2
```

**Comment by kbolino** [comment]
I tried downgrading to `v0.4.9` and that produced a *different* problem:

```console
> go build . 
# github.com/veandco/go-sdl2/ttf
..\..\..\go\pkg\mod\github.com\veandco\go-sdl2@v0.4.10\ttf\sdl_ttf.go:85:32: cannot use _Ctype_int(val) (type _Ctype_int) as type _Ctype_SDL_bool in argument to _Cfunc_TTF_ByteSwappedUNICODE
```

Ditto for `v0.4.10`. I was able to resolve that issue by also downgrading SDL2_ttf to `2.0.15` (the version I was using previously).

**Comment by veeableful** [comment]
Oh dear.. sorry for that. I tried the build using go-sdl2 v0.4.11 and SDL2-2.0.18 in Fedora Linux 35 and macOS Monterey and they worked for me. I will try to reproduce these issues if I am able and fix these hopefully by tomorrow.

**Comment by veeableful** [comment]
Ahh, I wasn't aware that there were new releases. It seems like SDL2 2.0.20 and SDL2_ttf 2.0.18 contained breaking API changes. I have made the fixes to v0.4.12. Hopefully, the issues are fixed!
